R/elasticsearch_parsers.R

```diff
@@ -402,46 +403,58 @@ unpack_nested_data <- function(chomped_df, col_to_unpack) {
         stop(msg)
     }
     
-    # Avoid side effects
     outDT <- data.table::copy(chomped_df)
-    
-    # Get the column to unpack
     listDT <- outDT[[col_to_unpack]]
     
-    # Make each row a data.table
-    listDT <- lapply(listDT, data.table::as.data.table)
-    
-    # Remove the empty ones... important, due to data.table 1.10.4 bug
-    oldIDs <- which(sapply(listDT, nrow) != 0)
-    listDT <- listDT[oldIDs]
-    
-    # Bind them together with an ID to match to the other data
-    newDT <- data.table::rbindlist(listDT, fill = TRUE, idcol = TRUE)
-    
-    # If we tried to unpack an empty column, fail
-    if (nrow(newDT) == 0) {
+    # Check for empty column
+    if (all(lengths(listDT) == 0)) {
```

Comment:
1) TIL `lengths()`
2) out of curiosity, what does this new version catch that `nrow(newDT)` didn't? It's not obvious to me

Reply:
This fails faster, I like it

Reply:
Actually, I don't think this will fail if all the data.tables have 0 rows, just if they all have 0 columns. Was this intended?

Reply:
Oh wait this is definitely just checking `nrow == 0` in a weird way. I'll fix it.

Reply:
Good point. I fixed it.